examples/astro-blog-demo/src/components/blog/CommentBlock.tsx

```diff
@@ -0,0 +1,201 @@
+import { CommentAuthorAvatar } from "@/components/ui/blog/CommentAuthorAvatar";
+import { CommentAuthorName } from "@/components/ui/blog/CommentAuthorName";
+import { CommentContent } from "@/components/ui/blog/CommentContent";
+import { CommentDate } from "@/components/ui/blog/CommentDate";
+import { Button } from "@/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuPortal,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { useMember } from "@/integrations/members";
+import { cn } from "@/lib/utils";
+import { Blog } from "@wix/blog/components";
+import { EllipsisVerticalIcon, TrashIcon } from "lucide-react";
+import React from "react";
+
+const {
+  Post: { Comment },
+} = Blog;
+
+interface CommentBlockProps {
+  children?: React.ReactNode;
+  className?: string;
+  uiLocale: string;
+}
+
+/**
+ * Displays a single comment with author information, content, date, and actions.
+ * Handles nested replies and shows parent comment context when replying to a reply.
+ * Includes delete functionality for comment owners.
+ */
+export const CommentBlock = React.forwardRef<HTMLDivElement, CommentBlockProps>(
+  ({ children, className, uiLocale, ...props }, ref) => {
+    const { isAuthenticated, member } = useMember();
+    const { deleteComment, comment } = Comment.useCommentContext();
+    // @ts-expect-error TODO -- Make this cleaner
+    const isOwner = comment.author?.memberId === member?._id;
```

Comment:
Create a `Comment.Root` component which accepts the current member (`Member`) and does the is owner check for example inside it.